---
title: "CRISPR_GUARD_publicationv2"
author: "Matthew Coelho"
date: "12/10/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

---
title: "HiFiCas9_analysis"
author: "Matthew Coelho"
date: "12/6/2019"
output: html_document
---

# data wrangling with tidyverse

```{r add NGS data to the dataframe of GUARD sequences and parameters}

# R version 3.5.1

library(tidyverse)

# add NGS data to the dataframe of GUARD sequences and parameters

dataframe <- as.tibble(read.table("dataframe", fill=TRUE, header=TRUE, sep="\t", stringsAsFactors=FALSE))
stats_BE <- as.tibble(read.table("stats_BE", fill=TRUE, header=TRUE, sep="\t", stringsAsFactors=FALSE))
stats_Cas9 <- as.tibble(read.table("stats_Cas9", fill=TRUE, header=TRUE, sep="\t", stringsAsFactors=FALSE))
stats_HiFiCas9 <- as.tibble(read.table("stats_HiFiCas9_061219", fill=TRUE, header=TRUE, sep="\t", stringsAsFactors=FALSE))
join1 <- left_join(dataframe, stats_Cas9, by = c("Number", "Condition"))
join2 <- left_join(join1, stats_BE, by = c("Number", "Condition"))
join3 <- left_join(join2, stats_HiFiCas9, by = c("Number", "Condition"))
write.csv(join3, file="join.csv")

data <- join3

```

```{r filter out GUARD constructs with NGS read counts < 250}
# filter out GUARD constructs with NGS read counts < 250

data %>% group_by(Number, Condition) %>% select(Number, Condition, Reads.Cas9, Reads.BE, Reads.HiFiCas9) %>% arrange(Reads.Cas9)
data %>% group_by(Number, Condition) %>% select(Number, Condition, Reads.Cas9, Reads.BE, Reads.HiFiCas9) %>% arrange(desc(Reads.Cas9))

data %>% group_by(Number, Condition) %>% select(Number, Condition, Reads.Cas9, Reads.BE, Reads.HiFiCas9) %>% arrange(Reads.BE)
data %>% group_by(Number, Condition) %>% select(Number, Condition, Reads.Cas9, Reads.BE, Reads.HiFiCas9) %>% arrange(desc(Reads.BE))

data %>% group_by(Number, Condition) %>% select(Number, Condition, Reads.Cas9, Reads.BE, Reads.HiFiCas9) %>% arrange(Reads.HiFiCas9)
data %>% group_by(Number, Condition) %>% select(Number, Condition, Reads.Cas9, Reads.BE, Reads.HiFiCas9) %>% arrange(desc(Reads.HiFiCas9))

data_clean_Cas9 <- filter(data, Reads.Cas9 > 250)
data_clean_BE <- filter(data, Reads.BE > 250)
data_clean_HiFiCas9 <- filter(data, Reads.HiFiCas9 > 250)

```

```{r add the gRNA sequences to the tibble}
# add the gRNA sequences to the tibble

data_clean_Cas9 <- data_clean_Cas9 %>% 
group_by(Guide) %>%
mutate(GuideSequence = ifelse(Guide == "EMX1", "GAGTCCGAGCAGAAGAAGAA", ifelse(Guide == "FANCF", "GGAATCCCTTCTGCAGCACC", ifelse(Guide == "HEKsite1", "GGGAAAGACCCAGCATCCGT", ifelse(Guide == "HBB", "CTTGCCCCACAGGGCAGTAA", "not recognised")))))

data_clean_BE <- data_clean_BE %>% 
group_by(Guide) %>%
mutate(GuideSequence = ifelse(Guide == "EMX1", "GAGTCCGAGCAGAAGAAGAA", ifelse(Guide == "FANCF", "GGAATCCCTTCTGCAGCACC", ifelse(Guide == "HEKsite1", "GGGAAAGACCCAGCATCCGT", ifelse(Guide == "HBB", "CTTGCCCCACAGGGCAGTAA", "not recognised")))))

data_clean_HiFiCas9 <- data_clean_HiFiCas9 %>% 
group_by(Guide) %>%
mutate(GuideSequence = ifelse(Guide == "EMX1", "GAGTCCGAGCAGAAGAAGAA", ifelse(Guide == "FANCF", "GGAATCCCTTCTGCAGCACC", ifelse(Guide == "HEKsite1", "GGGAAAGACCCAGCATCCGT", ifelse(Guide == "HBB", "CTTGCCCCACAGGGCAGTAA", "not recognised")))))

```

```{r add the number of off-target mismatches to the tibble}
#add the number of off-target mismatches to the tibble

data_clean_Cas9 <- data_clean_Cas9 %>% mutate(Mismatches = sum((strsplit(GuideSequence, split=""))[[1]] != (strsplit(OffGuide.y, split=""))[[1]]))

data_clean_BE <- data_clean_BE %>% mutate(Mismatches = sum((strsplit(GuideSequence, split=""))[[1]] != (strsplit(OffGuide.y, split=""))[[1]]))

data_clean_HiFiCas9 <- data_clean_HiFiCas9 %>% mutate(Mismatches = sum((strsplit(GuideSequence, split=""))[[1]] != (strsplit(OffGuide.y, split=""))[[1]]))
  
```

```{r filter out two poorly represented off-target sites with low read number and minority of GUARDs represented, and then correct names of non-unique off-target IDs}
# filter out two poorly represented off-target sites with low read number and minority of GUARDs represented, and then correct names of non-unique off-target IDs 

data_clean_Cas9 <- data_clean_Cas9 %>% 
mutate(ID = ifelse(OffGuide.y == "GAGTCCTAGCAGGAGAAGAA", "OT0009.EMX1", ID))

data_clean_BE <- data_clean_BE %>% 
mutate(ID = ifelse(OffGuide.y == "GAGTCCTAGCAGGAGAAGAA", "OT0009.EMX1", ID))

data_clean_HiFiCas9 <- data_clean_HiFiCas9 %>%
mutate(ID = ifelse(OffGuide.y == "GAGTCCTAGCAGGAGAAGAA", "OT0009.EMX1", ID))


data_clean_Cas9 <- data_clean_Cas9 %>%
filter(ID !="OT0005") %>%
filter(ID !="OT0009.EMX1")

data_clean_BE <- data_clean_BE %>%
filter(ID !="OT0005") %>%
filter(ID !="OT0009.EMX1")

data_clean_HiFiCas9 <- data_clean_HiFiCas9 %>%
filter(ID !="OT0005") %>%
filter(ID !="OT0009.EMX1")


data_clean_Cas9 <- data_clean_Cas9 %>%
mutate(ID = str_c(ID, Guide, sep="_")) %>%
mutate(ID = str_c(ID, "\n", "Mismatches = ", Mismatches, sep=" "))

data_clean_BE <- data_clean_Cas9 %>%
mutate(ID = str_c(ID, Guide, sep="_")) %>%
mutate(ID = str_c(ID, "\n", "Mismatches = ", Mismatches, sep=" "))

data_clean_HiFiCas9 <- data_clean_HiFiCas9 %>%
mutate(ID = str_c(ID, Guide, sep="_")) %>%
mutate(ID = str_c(ID, "\n", "Mismatches = ", Mismatches, sep=" "))

```

```{r change labelling of experimental conditions}
# change labelling of experimental conditions

data_clean_Cas9 <- data_clean_Cas9 %>%
mutate(Condition = ifelse(Condition == "C1", "Control1", ifelse(Condition == "C2", "Control2", ifelse(Condition == "D1", "Dox.1", ifelse(Condition == "D2", "Dox.2", ifelse(Condition == "Plasmid", "Plasmid", Condition))))))

data_clean_BE <- data_clean_Cas9 %>%
mutate(Condition = ifelse(Condition == "C1", "Control1", ifelse(Condition == "C2", "Control2", ifelse(Condition == "D1", "Dox.1", ifelse(Condition == "D2", "Dox.2", ifelse(Condition == "Plasmid", "Plasmid", Condition))))))

data_clean_HiFiCas9 <- data_clean_HiFiCas9 %>%
mutate(Condition = ifelse(Condition == "C1", "Control1", ifelse(Condition == "C2", "Control2", ifelse(Condition == "D1", "Dox.1", ifelse(Condition == "D2", "Dox.2", ifelse(Condition == "Plasmid", "Plasmid", Condition))))))
  
```

```{r find GUARDs that are "Functional" i.e. protect from editing > 2sd from average of non-targeting (NT) GUARDs for off-target}
# find GUARDs that are "Functional" i.e. protect from editing > 2sd from average of non-targeting (NT) GUARDs for off-target

NT_benchmarks_Cas9 <- data_clean_Cas9 %>% 
filter(NT=="TRUE") %>%
filter(Condition == "Dox.1" | Condition == "Dox.2") %>% 
group_by(ID) %>%
summarise(mean_NT = mean(X.Indels.Cas9), sd_NT = sd(X.Indels.Cas9))

join_Functional_GUARDs <- left_join(data_clean_Cas9, NT_benchmarks_Cas9, by = c("ID"))
data_clean_Cas9_split <- join_Functional_GUARDs %>% 
filter(Condition == "Dox.1"|Condition =="Dox.2") %>%
group_by(ID) %>% 
mutate(Functional = ifelse(X.Indels.Cas9 < (mean_NT - 2*sd_NT) & NT == "FALSE", "TRUE", "FALSE"))

#join_Functional_GUARDs2 <- left_join(data_clean_HiFiCas9, NT_benchmarks_Cas9, by = c("ID"))
#data_clean_HiFiCas9_split <- join_Functional_GUARDs2 %>% 
#filter(Condition == "Dox.1"|Condition =="Dox.2") %>%
#group_by(ID) %>% 
#mutate(Functional = ifelse(X.Indels.Cas9 < (mean_NT - 2*sd_NT) & NT == "FALSE", "TRUE", "FALSE"))

NT_benchmarks_HiFiCas9 <- data_clean_HiFiCas9 %>% 
filter(NT=="TRUE") %>%
filter(Condition == "Dox.1" | Condition == "Dox.2") %>% 
group_by(ID) %>%
summarise(mean_NT = mean(X.Indels.HiFiCas9), sd_NT = sd(X.Indels.HiFiCas9))

join_Functional_GUARDsHiFi <- left_join(data_clean_HiFiCas9, NT_benchmarks_HiFiCas9, by = c("ID"))
data_clean_HiFiCas9_split <- join_Functional_GUARDsHiFi %>% 
filter(Condition == "Dox.1"|Condition =="Dox.2") %>%
group_by(ID) %>% 
mutate(Functional = ifelse(X.Indels.HiFiCas9 < (mean_NT - 2*sd_NT) & NT == "FALSE", "TRUE", "FALSE"))


NT_benchmarks_BE <- data_clean_BE %>% 
filter(NT=="TRUE") %>%
filter(Condition == "Dox.1" | Condition == "Dox.2") %>% 
group_by(ID) %>%
summarise(mean_NT = mean(X.SNPs.BE), sd_NT = sd(X.SNPs.BE))

join_Functional_GUARDs_BE <- left_join(data_clean_BE, NT_benchmarks_BE, by = c("ID"))
data_clean_BE_split <- join_Functional_GUARDs_BE %>% 
filter(Condition == "Dox.1"|Condition =="Dox.2") %>%
group_by(ID) %>% 
mutate(Functional = ifelse(X.SNPs.BE < (mean_NT - 2*sd_NT) & NT == "FALSE", "TRUE", "FALSE"))

```

```{r define HiFi GUARDs with near perfect protection}
# define HiFi GUARDs with near perfect protection

data_clean_HiFiCas9_split <- data_clean_HiFiCas9_split %>%
  mutate(Perfect = ifelse(X.Indels.HiFiCas9 < 0.5 & NT == "FALSE" & mean_NT > 0.5, "TRUE", "FALSE"))

```

```{r define GUARDs with cystosine at each position for BE GUARD RNA screen}
# define GUARDs with cystosine at each position for BE GUARD RNA screen

# find out if BE editing GUARDs have a C as 15 nt of the GUARD sequence

data_clean_BE_split <- data_clean_BE_split %>% mutate(C_in_15 = grepl("C", substr(GuardWithPAM, 15, 15)))

# find out if BE editing GUARDs have a C within the 14 bases of the GUARD sequence

data_clean_BE_split <- data_clean_BE_split %>% mutate(C_in_14 = grepl("C", substr(GuardWithPAM, 14, 14)))

# find out if BE editing GUARDs have a C within the 13 bases of the GUARD sequence

data_clean_BE_split <- data_clean_BE_split %>% mutate(C_in_13 = grepl("C", substr(GuardWithPAM, 13, 13)))

# find out if BE editing GUARDs have a C within the 12 bases of the GUARD sequence

data_clean_BE_split <- data_clean_BE_split %>% mutate(C_in_12 = grepl("C", substr(GuardWithPAM, 12, 12)))

# find out if BE editing GUARDs have a C within the 11 bases of the GUARD sequence

data_clean_BE_split <- data_clean_BE_split %>% mutate(C_in_11 = grepl("C", substr(GuardWithPAM, 11, 11)))

# find out if BE editing GUARDs have a C within the 10 bases of the GUARD sequence

data_clean_BE_split <- data_clean_BE_split %>% mutate(C_in_10 = grepl("C", substr(GuardWithPAM, 10, 10)))

# find out if BE editing GUARDs have a C within the 9 bases of the GUARD sequence

data_clean_BE_split <- data_clean_BE_split %>% mutate(C_in_9 = grepl("C", substr(GuardWithPAM, 9, 9)))

# find out if BE editing GUARDs have a C within the 8 bases of the GUARD sequence

data_clean_BE_split <- data_clean_BE_split %>% mutate(C_in_8 = grepl("C", substr(GuardWithPAM, 8, 8)))

# find out if BE editing GUARDs have a C within the 7 bases of the GUARD sequence

data_clean_BE_split <- data_clean_BE_split %>% mutate(C_in_7 = grepl("C", substr(GuardWithPAM, 7, 7)))

# find out if BE editing GUARDs have a C within the 6 bases of the GUARD sequence

data_clean_BE_split <- data_clean_BE_split %>% mutate(C_in_6 = grepl("C", substr(GuardWithPAM, 6, 6)))

# find out if BE editing GUARDs have a C within the 5 bases of the GUARD sequence

data_clean_BE_split <- data_clean_BE_split %>% mutate(C_in_5 = grepl("C", substr(GuardWithPAM, 5, 5)))

# find out if BE editing GUARDs have a C within the 4 bases of the GUARD sequence

data_clean_BE_split <- data_clean_BE_split %>% mutate(C_in_4 = grepl("C", substr(GuardWithPAM, 4, 4)))

# find out if BE editing GUARDs have a C within the 3 bases of the GUARD sequence

data_clean_BE_split <- data_clean_BE_split %>% mutate(C_in_3 = grepl("C", substr(GuardWithPAM, 3, 3)))

# find out if BE editing GUARDs have a C within the 2 bases of the GUARD sequence

data_clean_BE_split <- data_clean_BE_split %>% mutate(C_in_2 = grepl("C", substr(GuardWithPAM, 2, 2)))

# find out if BE editing GUARDs have a C within the 1 bases of the GUARD sequence

data_clean_BE_split <- data_clean_BE_split %>% mutate(C_in_1 = grepl("C", substr(GuardWithPAM, 1, 1)))

```

```{r add Doench et al. prediction "Efficacy" and "Confidence" stats on short GUARDs}
# add Doench et al. prediction "Efficacy" and "Confidence" stats on short GUARDs

Efficacy_stats <- as.tibble(read.table("Efficacy_stats", fill=TRUE, header=TRUE, sep="\t", stringsAsFactors=FALSE)) %>% select(Number, Efficacy, Confidence)
join <- left_join(data_clean_Cas9_split, Efficacy_stats, by = c("Number"))
join2 <- left_join(data_clean_BE_split, Efficacy_stats, by = c("Number"))
join3 <- left_join(data_clean_HiFiCas9_split, Efficacy_stats, by = c("Number"))

data_clean_Cas9_split <- join
data_clean_BE_split <- join2
data_clean_HiFiCas9_split <- join3

```

```{r add a score ("Protection") of functionality for the GUARDs based on average off-target editing. Normalises the GAURD protection score for different levels of editing for each off-target}

# add a score ("Protection") of functionality for the GUARDs based on average off-target editing. Normalises the GAURD protection score for different levels of editing for each off-target

data_clean_Cas9_split <- data_clean_Cas9_split %>% mutate(Protection=((mean_NT-X.Indels.Cas9)/(mean_NT)))

data_clean_HiFiCas9_split <- data_clean_HiFiCas9_split %>% mutate(Protection=((mean_NT-X.Indels.HiFiCas9)/(mean_NT)))

data_clean_BE_split <- data_clean_BE_split %>% mutate(Protection=((mean_NT-X.SNPs.BE)/(mean_NT)))

```

```{r write final output as a csv}
# write final output as a csv

write.csv(data_clean_Cas9_split, file = "data_clean_Cas9_split.csv")
write.csv(data_clean_BE_split, file = "data_clean_BE_split.csv")
write.csv(data_clean_HiFiCas9_split, file = "data_clean_HiFiCas9_split.csv")

```

#############################################################################################################################################

# making plots and statistics for plots with tidyverse and ggplot2

```{r correlation between replicates Cas9}
# correlation between replicates Cas9

#Cas9
#control

C1 <- filter(data_clean_Cas9, Condition == "Control1") %>% group_by(Number) %>% select(X.Indels.Cas9, Number)
C2 <- filter(data_clean_Cas9, Condition == "Control2") %>% group_by(Number) %>% select(X.Indels.Cas9, Number)
join_C1_C2 <- left_join(C1, C2, by =c("Number"))

summary(lm(join_C1_C2$X.Indels.Cas9.x ~ join_C1_C2$X.Indels.Cas9.y))

```

```{r}

plot_reps_Cas9C <- ggplot(join_C1_C2, aes(x=X.Indels.Cas9.x, y=X.Indels.Cas9.y)) + 
geom_jitter(size=3, alpha=0.8, width = 0.2, height = 0.2) + labs(title= ("Cas9 \nAdj. R-squared: 0.4216"), x="Control replicate 1: Indels (%) \n n = 602", y="Control replicate 2: Indels (%) \n n = 600") + 
geom_smooth(method = lm) +
ylim(0,100) + xlim(0,100)
plot_reps_Cas9C

```

```{r}
#dox

D1 <- filter(data_clean_Cas9, Condition == "Dox.1") %>% group_by(Number) %>% select(X.Indels.Cas9, Number)
D2 <- filter(data_clean_Cas9, Condition == "Dox.2") %>% group_by(Number) %>% select(X.Indels.Cas9, Number)
join_D1_D2 <- left_join(D1, D2, by =c("Number"))

summary(lm(join_D1_D2$X.Indels.Cas9.x ~ join_D1_D2$X.Indels.Cas9.y))

```

```{r}

plot_reps_Cas9D <- ggplot(join_D1_D2, aes(x=X.Indels.Cas9.x, y=X.Indels.Cas9.y)) + 
geom_jitter(size=3, alpha=0.8, width = 0.2, height = 0.2) + labs(title= ("Cas9 \nAdj. R-squared: 0.9758"), x="Dox. replicate 1: Indels (%) \n n = 600", y="Dox. replicate 2: Indels (%) \n n = 601") + 
geom_smooth(method = lm) +
ylim(0,100) + xlim(0,100)
plot_reps_Cas9D

```

```{r correlation between replicates HiFiCas9}
# correlation between replicates HiFiCas9

#HiFiCas9
#control

HiFiC1 <- filter(data_clean_HiFiCas9, Condition == "Control1") %>% group_by(Number) %>% select(X.Indels.HiFiCas9, Number)
HiFiC2 <- filter(data_clean_HiFiCas9, Condition == "Control2") %>% group_by(Number) %>% select(X.Indels.HiFiCas9, Number)
join_HiFiC1_HiFiC2 <- left_join(HiFiC1, HiFiC2, by =c("Number"))

summary(lm(join_HiFiC1_HiFiC2$X.Indels.HiFiCas9.x ~ join_HiFiC1_HiFiC2$X.Indels.HiFiCas9.y))

```

```{r}

plot_reps_HiFiCas9C <- ggplot(join_HiFiC1_HiFiC2, aes(x=X.Indels.HiFiCas9.x, y=X.Indels.HiFiCas9.y)) + 
geom_jitter(size=3, alpha=0.8, width = 0.2, height = 0.2) + labs(title= ("HiFi-Cas9 \nAdj. R-squared: 0.5044"), x="Control replicate 1: Indels (%) \n n = 602", y="Control replicate 2: Indels (%) \n n = 600") + 
geom_smooth(method = lm) +
ylim(0,100) + xlim(0,100)
plot_reps_HiFiCas9C

```

```{r}
#dox

HiFiD1 <- filter(data_clean_HiFiCas9, Condition == "Dox.1") %>% group_by(Number) %>% select(X.Indels.HiFiCas9, Number)
HiFiD2 <- filter(data_clean_HiFiCas9, Condition == "Dox.2") %>% group_by(Number) %>% select(X.Indels.HiFiCas9, Number)
join_HiFiD1_HiFiD2 <- left_join(HiFiD1, HiFiD2, by =c("Number"))

summary(lm(join_HiFiD1_HiFiD2$X.Indels.HiFiCas9.x ~ join_HiFiD1_HiFiD2$X.Indels.HiFiCas9.y))

```

```{r}

plot_reps_HiFiCas9D <- ggplot(join_HiFiD1_HiFiD2, aes(x=X.Indels.HiFiCas9.x, y=X.Indels.HiFiCas9.y)) + 
geom_jitter(size=3, alpha=0.8, width = 0.2, height = 0.2) + labs(title= ("HiFi-Cas9 \nAdj. R-squared: 0.5315"), x="Dox. replicate 1: Indels (%) \n n = 600", y="Dox. replicate 2: Indels (%) \n n = 601") + 
geom_smooth(method = lm) +
ylim(0,100) + xlim(0,100)
plot_reps_HiFiCas9D

```

```{r correlation between replicates BE}
# correlation between replicates BE

#BE
#control

C1BE <- filter(data_clean_BE, Condition == "Control1") %>% group_by(Number) %>% select(X.SNPs.BE, Number)
C2BE <- filter(data_clean_BE, Condition == "Control2") %>% group_by(Number) %>% select(X.SNPs.BE, Number)
join_C1_C2_BE <- left_join(C1BE, C2BE, by =c("Number"))

summary(lm(join_C1_C2_BE$X.SNPs.BE.x ~ join_C1_C2_BE$X.SNPs.BE.y))

```

```{r}

plot_reps_BEC <- ggplot(join_C1_C2_BE, aes(x=X.SNPs.BE.x, y=X.SNPs.BE.y)) + 
geom_jitter(size=3, alpha=0.8, width = 0.2, height = 0.2) + labs(title= ("BE3 \nAdj. R-squared: 0.4695"), x="Control replicate 1: BE (%) \n n = 602", y="Control replicate 2: BE (%)\n n = 600") + 
geom_smooth(method = lm) +
ylim(0,100) + xlim(0,100)
plot_reps_BEC

```

```{r}

#dox

D1BE <- filter(data_clean_BE, Condition == "Dox.1") %>% group_by(Number) %>% select(X.SNPs.BE, Number)
D2BE <- filter(data_clean_BE, Condition == "Dox.2") %>% group_by(Number) %>% select(X.SNPs.BE, Number)
join_D1_D2_BE <- left_join(D1BE, D2BE, by =c("Number"))

summary(lm(join_D1_D2_BE$X.SNPs.BE.x ~ join_D1_D2_BE$X.SNPs.BE.y))

```

```{r}

plot_reps_BED <- ggplot(join_D1_D2_BE, aes(x=X.SNPs.BE.x, y=X.SNPs.BE.y)) + 
geom_jitter(size=3, alpha=0.8, width = 0.2, height = 0.2) + labs(title= ("BE3 \nAdj. R-squared: 0.9792"), x="Dox. replicate 1: BE (%)\n n = 600", y="Dox. replicate 2: BE (%)\n n = 601") + 
geom_smooth(method = lm) +
ylim(0,100) + xlim(0,100)
plot_reps_BED

```

```{r quantify the number of GUARD RNAs represented in each group}
# quantify the number of GUARD RNAs represented in each group

data_clean_Cas9 %>%
group_by(Condition, Number) %>%
filter(Condition == "Control1") %>%
tally() %>% 
count(n)

data_clean_Cas9 %>%
group_by(Condition, Number) %>%
filter(Condition == "Control2") %>%
tally() %>% 
count(n)

data_clean_Cas9 %>%
group_by(Condition, Number) %>%
filter(Condition == "Dox.1") %>%
tally() %>% 
count(n)

data_clean_Cas9 %>%
group_by(Condition, Number) %>%
filter(Condition == "Dox.2") %>%
tally() %>% 
count(n)

####

data_clean_HiFiCas9 %>%
group_by(Condition, Number) %>%
filter(Condition == "Control1") %>%
tally() %>% 
count(n)

data_clean_HiFiCas9 %>%
group_by(Condition, Number) %>%
filter(Condition == "Control2") %>%
tally() %>% 
count(n)

data_clean_HiFiCas9 %>%
group_by(Condition, Number) %>%
filter(Condition == "Dox.1") %>%
tally() %>% 
count(n)

data_clean_HiFiCas9 %>%
group_by(Condition, Number) %>%
filter(Condition == "Dox.2") %>%
tally() %>% 
count(n)

####

data_clean_BE %>%
group_by(Condition, Number) %>%
filter(Condition == "Control1") %>%
tally() %>% 
count(n)

data_clean_BE %>%
group_by(Condition, Number) %>%
filter(Condition == "Control2") %>%
tally() %>% 
count(n)

data_clean_Cas9 %>%
group_by(Condition, Number) %>%
filter(Condition == "Dox.1") %>%
tally() %>% 
count(n)

data_clean_BE %>%
group_by(Condition, Number) %>%
filter(Condition == "Dox.2") %>%
tally() %>% 
count(n)

```

```{r stats for Cas9 indels between conditions}
# stats for Cas9 indels between conditions

mean_C1 <- data_clean_Cas9 %>%  
filter(Condition == "Control1") %>%
as.data.frame(mean_C1)

mean_C2 <- data_clean_Cas9 %>%  
filter(Condition == "Control2") %>%
as.data.frame(mean_C2)

mean_D1 <- data_clean_Cas9 %>% 
filter(Condition == "Dox.1") %>%
as.data.frame(mean_D1)

mean_D2 <- data_clean_Cas9 %>% 
filter(Condition == "Dox.2") %>%
as.data.frame(mean_D2)

t.test(mean_C1$X.Indels.Cas9, mean_D1$X.Indels.Cas9, var.equal = TRUE)
t.test(mean_C1$X.Indels.Cas9, mean_D2$X.Indels.Cas9, var.equal = TRUE)
t.test(mean_C2$X.Indels.Cas9, mean_D1$X.Indels.Cas9, var.equal = TRUE)
t.test(mean_C2$X.Indels.Cas9, mean_D2$X.Indels.Cas9, var.equal = TRUE)

t.test(mean_C1$X.Indels.Cas9, mean_C2$X.Indels.Cas9, var.equal = TRUE)
t.test(mean_D1$X.Indels.Cas9, mean_D2$X.Indels.Cas9, var.equal = TRUE)

```

```{r plot Cas9 indels in each condition}
# plot Cas9 indels in each condition

plot_condition_Cas9 <- ggplot(subset(data_clean_Cas9, Condition != "Plasmid"), aes(x=Condition, y=X.Indels.Cas9)) +
geom_boxplot() + 
labs(title= "Cas9: Off-target Indel frequency", x=NULL, y="Indels (%)") +
ylim(-3, 100) +
annotate("text", x = 1.5:3.5, y = 98, label = c("N.S", "P= 2.2x10-16", "N.S")) +
annotate("text", x = 1:4, y = -2.6, label = c("italic(n) == 602", "italic(n) == 600", "italic(n) == 600", "italic(n) == 601"), parse = TRUE) +
annotate("segment", x = 1.1, xend = 1.9, y = 95, yend = 95, alpha = 0.7) +
annotate("segment", x = 2.1, xend = 2.9, y = 95, yend = 95, alpha = 0.7) +
annotate("segment", x = 3.1, xend = 3.9, y = 95, yend = 95, alpha = 0.7) +
theme(axis.text.x = element_text(size = rel(1.2)))
plot_condition_Cas9

```

```{r stats for HiFiCas9 indels between conditions}
# stats for HiFiCas9 indels between conditions

mean_HiFiC1 <- data_clean_HiFiCas9 %>%  
filter(Condition == "Control1") %>%
as.data.frame(mean_HiFiC1)

mean_HiFiC2 <- data_clean_HiFiCas9 %>%  
filter(Condition == "Control2") %>%
as.data.frame(mean_HiFiC2)

mean_HiFiD1 <- data_clean_HiFiCas9 %>% 
filter(Condition == "Dox.1") %>%
as.data.frame(mean_HiFiD1)

mean_HiFiD2 <- data_clean_HiFiCas9 %>% 
filter(Condition == "Dox.2") %>%
as.data.frame(mean_HiFiD2)

t.test(mean_HiFiC1$X.Indels.HiFiCas9, mean_HiFiD1$X.Indels.HiFiCas9, var.equal = TRUE)
t.test(mean_HiFiC1$X.Indels.HiFiCas9, mean_HiFiD2$X.Indels.HiFiCas9, var.equal = TRUE)
t.test(mean_HiFiC2$X.Indels.HiFiCas9, mean_HiFiD1$X.Indels.HiFiCas9, var.equal = TRUE)
t.test(mean_HiFiC2$X.Indels.HiFiCas9, mean_HiFiD2$X.Indels.HiFiCas9, var.equal = TRUE)

t.test(mean_HiFiC1$X.Indels.HiFiCas9, mean_HiFiC2$X.Indels.HiFiCas9, var.equal = TRUE)
t.test(mean_HiFiD1$X.Indels.HiFiCas9, mean_HiFiD2$X.Indels.HiFiCas9, var.equal = TRUE)

```

```{r plot HiFiCas9 indels in each condition}
# plot HiFiCas9 indels in each condition

plot_condition_HiFiCas9 <- ggplot(data_clean_HiFiCas9, aes(x=Condition, y=X.Indels.HiFiCas9)) +
geom_boxplot() + 
labs(title= "HiFi-Cas9: Off-target Indel frequency", x=NULL, y="Indels (%)") +
ylim(-3, 100) +
annotate("text", x = 1.5:3.5, y = 98, label = c("N.S", "N.S", "N.S")) +
annotate("text", x = 1:4, y = -2.6, label = c("italic(n) == 567", "italic(n) == 541", "italic(n) == 563", "italic(n) == 564"), parse = TRUE) +
annotate("segment", x = 1.1, xend = 1.9, y = 95, yend = 95, alpha = 0.7) +
annotate("segment", x = 2.1, xend = 2.9, y = 95, yend = 95, alpha = 0.7) +
annotate("segment", x = 3.1, xend = 3.9, y = 95, yend = 95, alpha = 0.7) +
theme(axis.text.x = element_text(size = rel(1.2)))
plot_condition_HiFiCas9

```

```{r stats for BE SNPs in each condition}
# stats for BE SNPs in each condition

mean_C1_BE <- data_clean_BE %>% 
filter(Condition == "Control1") %>%
as.data.frame(mean_C1_BE)

mean_C2_BE <- data_clean_BE %>%  
filter(Condition == "Control2") %>%
as.data.frame(mean_C2_BE)

mean_D1_BE <- data_clean_BE %>%  
filter(Condition == "Dox.1") %>%
as.data.frame(mean_D1_BE)

mean_D2_BE <- data_clean_BE %>%  
filter(Condition == "Dox.2") %>%
as.data.frame(mean_D2_BE)

t.test(mean_C1_BE$X.SNPs.BE, mean_D1_BE$X.SNPs.BE, var.equal = TRUE)
t.test(mean_C1_BE$X.SNPs.BE, mean_D2_BE$X.SNPs.BE, var.equal = TRUE)
t.test(mean_C2_BE$X.SNPs.BE, mean_D1_BE$X.SNPs.BE, var.equal = TRUE)
t.test(mean_C2_BE$X.SNPs.BE, mean_D2_BE$X.SNPs.BE, var.equal = TRUE)

t.test(mean_C1_BE$X.SNPs.BE, mean_C2_BE$X.SNPs.BE, var.equal = TRUE)
t.test(mean_D1_BE$X.SNPs.BE, mean_D2_BE$X.SNPs.BE, var.equal = TRUE)

```

```{r plot BE SNPs per group}
# plot BE SNPs per group

plot_condition_BE <- ggplot(subset(data_clean_BE, Condition != "Plasmid"), aes(x=Condition, y=X.SNPs.BE)) +
geom_boxplot() + 
labs(title= "BE: Off-target Base Edits", x=NULL, y="BE (%)") +
ylim(-3, 100) +
annotate("text", x = 1.5:3.5, y = 98, label = c("P= 5.54x10-5", "P= 2.2x10-16", "N.S")) +
annotate("text", x = 1:4, y = -2.6, label = c("italic(n) == 602", "italic(n) == 600", "italic(n) == 600", "italic(n) == 601"), parse = TRUE) +
annotate("segment", x = 1.1, xend = 1.9, y = 95, yend = 95, alpha = 0.7) +
annotate("segment", x = 2.1, xend = 2.9, y = 95, yend = 95, alpha = 0.7) +
annotate("segment", x = 3.1, xend = 3.9, y = 95, yend = 95, alpha = 0.7) +
theme(axis.text.x = element_text(size = rel(1.2)))
plot_condition_BE

```

```{r NT stats Cas9}
# NT stats Cas9

mean_DNTTRUEC <- data_clean_Cas9 %>% 
filter(Condition == "Dox.1" | Condition =="Dox.2") %>%
filter(NT == "TRUE") %>%
as.data.frame(mean_DNTTRUEC)

mean_DNTFALSEC <- data_clean_Cas9 %>% 
filter(Condition == "Dox.1" | Condition =="Dox.2") %>%
filter(NT == "FALSE") %>%
as.data.frame(mean_DNTFALSEC)

t.test(mean_DNTTRUEC$X.Indels.Cas9, mean_DNTFALSEC$X.Indels.Cas9, var.equal = TRUE)

```

```{r NT plot Cas9}
# NT plot Cas9

plot_NT_Cas9box <- ggplot(subset(data_clean_Cas9, Condition == "Dox.1" | Condition == "Dox.2"), aes(x=NT, y=X.Indels.Cas9)) +
geom_boxplot(colour = "Red") +
  theme_bw() +
geom_jitter(alpha=0.1, size = 2, width = 0.2) +
  labs(title= "Cas9: Off-target Indels", y="Indels (%)", x=NULL) +
annotate("text", x = 1.5, y = 20, label = c("P= 0.08148")) +
annotate("segment", x = 1.1, xend = 1.9, y = 19.4, yend = 19.4, alpha = 0.7) +
coord_cartesian(ylim=c(0, 20)) +
scale_x_discrete(labels=c("FALSE" = "GUARD RNAs", "TRUE" = "NT controls")) +
theme(axis.text.x = element_text(size = rel(1.2)))
plot_NT_Cas9box

```

```{r NT stats HiFiCas9}
# NT stats HiFiCas9

mean_DNTTRUEC_HiFi <- data_clean_HiFiCas9 %>% 
filter(Condition == "Dox.1" | Condition =="Dox.2") %>%
filter(NT == "TRUE") %>%
as.data.frame(mean_DNTTRUEC_HiFi)

mean_DNTFALSEC_HiFi <- data_clean_HiFiCas9 %>% 
filter(Condition == "Dox.1" | Condition =="Dox.2") %>%
filter(NT == "FALSE") %>%
as.data.frame(mean_DNTFALSEC_HiFi)

t.test(mean_DNTTRUEC_HiFi$X.Indels.HiFiCas9, mean_DNTFALSEC_HiFi$X.Indels.HiFiCas9, var.equal = TRUE)

```

```{r NT plot HiFiCas9}
# NT plot HiFiCas9

plot_NT_HiFiCas9box <- ggplot(subset(data_clean_HiFiCas9, Condition == "Dox.1" | Condition == "Dox.2"), aes(x=NT, y=X.Indels.HiFiCas9)) +
geom_boxplot(colour = "Red") +
  theme_bw() +
geom_jitter(alpha=0.1, size = 2, width = 0.2) +
  labs(title= "HiFi-Cas9: Off-target Indels", y="Indels (%)", x=NULL) +
annotate("text", x = 1.5, y = 20, label = c("N.S")) +
annotate("segment", x = 1.1, xend = 1.9, y = 19.4, yend = 19.4, alpha = 0.7) +
coord_cartesian(ylim=c(0, 20)) +
scale_x_discrete(labels=c("FALSE" = "GUARD RNAs", "TRUE" = "NT controls")) +
theme(axis.text.x = element_text(size = rel(1.2)))
plot_NT_HiFiCas9box

```

```{r NT stats BE}
# NT stats BE

mean_NT_TRUE_BE <- data_clean_BE %>% 
filter(Condition == "Dox.1" | Condition == "Dox.2") %>%
filter(NT == "TRUE") %>%
as.data.frame(mean_NT_TRUE_BE)

mean_NT_FALSE_BE <- data_clean_BE %>% 
filter(Condition == "Dox.1" | Condition == "Dox.2") %>%
filter(NT == "FALSE") %>%
as.data.frame(mean_NT_FALSE_BE)

t.test(mean_NT_TRUE_BE$X.SNPs.BE, mean_NT_FALSE_BE$X.SNPs.BE, var.equal = TRUE)

```

```{r NT plot BE}
# NT plot BE

plot_NT_BEbox <- ggplot(subset(data_clean_BE, Condition == "Dox.1" | Condition == "Dox.2"), aes(x=NT, y=X.SNPs.BE)) +
geom_boxplot(colour = "Red") +
  theme_bw() +
geom_jitter(alpha=0.1, size = 2, width = 0.2) +
  labs(title= "BE: Off-target Base Editing", x=NULL, y="BE (%)") +
annotate("text", x = 1.5, y = 98, label = c("P= 7.899x10-8")) +
annotate("segment", x = 1.1, xend = 1.9, y = 95, yend = 95, alpha = 0.7)+
scale_x_discrete(labels=c("FALSE" = "GUARD RNAs", "TRUE" = "NT controls")) +
theme(axis.text.x = element_text(size = rel(1.2)))
plot_NT_BEbox

```

```{r stats for cytosines at each position in BE GUARD RNAs}
# stats for cytosines at each position in BE GUARD RNAs

# C in 15 bases from PAM in BE GUARDs
#stats

mean_C_in_15_TRUE <- data_clean_BE_split %>% 
filter(C_in_15 == "TRUE") %>%
as.data.frame(mean_C_C_in_15_TRUE)

mean_C_in_15_FALSE <- data_clean_BE_split %>%
filter(C_in_15 == "FALSE") %>%
as.data.frame(mean_C_in_15_FALSE)

t.test(mean_C_in_15_TRUE$X.SNPs.BE, mean_C_in_15_FALSE$X.SNPs.BE, var.equal = TRUE)

# C in 14 bases from PAM in BE GUARDs
#stats

mean_C_in_14_TRUE <- data_clean_BE_split %>% 
filter(C_in_14 == "TRUE") %>%
as.data.frame(mean_C_C_in_14_TRUE)

mean_C_in_14_FALSE <- data_clean_BE_split %>%
filter(C_in_14 == "FALSE") %>%
as.data.frame(mean_C_in_14_FALSE)

t.test(mean_C_in_14_TRUE$X.SNPs.BE, mean_C_in_14_FALSE$X.SNPs.BE, var.equal = TRUE)

# C in 13 bases from PAM in BE GUARDs
#stats

mean_C_in_13_TRUE <- data_clean_BE_split %>% 
filter(C_in_13 == "TRUE") %>%
as.data.frame(mean_C_C_in_13_TRUE)

mean_C_in_13_FALSE <- data_clean_BE_split %>%
filter(C_in_13 == "FALSE") %>%
as.data.frame(mean_C_in_13_FALSE)

t.test(mean_C_in_13_TRUE$X.SNPs.BE, mean_C_in_13_FALSE$X.SNPs.BE, var.equal = TRUE)

# C in 12 bases from PAM in BE GUARDs
#stats

mean_C_in_12_TRUE <- data_clean_BE_split %>% 
filter(C_in_12 == "TRUE") %>%
as.data.frame(mean_C_C_in_12_TRUE)

mean_C_in_12_FALSE <- data_clean_BE_split %>%
filter(C_in_12 == "FALSE") %>%
as.data.frame(mean_C_in_12_FALSE)

t.test(mean_C_in_12_TRUE$X.SNPs.BE, mean_C_in_12_FALSE$X.SNPs.BE, var.equal = TRUE)

# C in 11 bases from PAM in BE GUARDs
#stats

mean_C_in_11_TRUE <- data_clean_BE_split %>% 
filter(C_in_11 == "TRUE") %>%
as.data.frame(mean_C_C_in_11_TRUE)

mean_C_in_11_FALSE <- data_clean_BE_split %>%
filter(C_in_11 == "FALSE") %>%
as.data.frame(mean_C_in_11_FALSE)

t.test(mean_C_in_11_TRUE$X.SNPs.BE, mean_C_in_11_FALSE$X.SNPs.BE, var.equal = TRUE)

# C in 10 bases from PAM in BE GUARDs
#stats

mean_C_in_10_TRUE <- data_clean_BE_split %>% 
filter(C_in_10 == "TRUE") %>%
as.data.frame(mean_C_C_in_10_TRUE)

mean_C_in_10_FALSE <- data_clean_BE_split %>%
filter(C_in_10 == "FALSE") %>%
as.data.frame(mean_C_in_10_FALSE)

t.test(mean_C_in_10_TRUE$X.SNPs.BE, mean_C_in_10_FALSE$X.SNPs.BE, var.equal = TRUE)

# C in 9 bases from PAM in BE GUARDs
#stats

mean_C_in_9_TRUE <- data_clean_BE_split %>% 
filter(C_in_9 == "TRUE") %>%
as.data.frame(mean_C_C_in_9_TRUE)

mean_C_in_9_FALSE <- data_clean_BE_split %>%
filter(C_in_9 == "FALSE") %>%
as.data.frame(mean_C_in_9_FALSE)

t.test(mean_C_in_9_TRUE$X.SNPs.BE, mean_C_in_9_FALSE$X.SNPs.BE, var.equal = TRUE)

# C in 8 bases from PAM in BE GUARDs
#stats

mean_C_in_8_TRUE <- data_clean_BE_split %>% 
filter(C_in_8 == "TRUE") %>%
as.data.frame(mean_C_C_in_8_TRUE)

mean_C_in_8_FALSE <- data_clean_BE_split %>%
filter(C_in_8 == "FALSE") %>%
as.data.frame(mean_C_in_8_FALSE)

t.test(mean_C_in_8_TRUE$X.SNPs.BE, mean_C_in_8_FALSE$X.SNPs.BE, var.equal = TRUE)

# C in 7 bases from PAM in BE GUARDs
#stats

mean_C_in_7_TRUE <- data_clean_BE_split %>% 
filter(C_in_7 == "TRUE") %>%
as.data.frame(mean_C_C_in_7_TRUE)

mean_C_in_7_FALSE <- data_clean_BE_split %>%
filter(C_in_7 == "FALSE") %>%
as.data.frame(mean_C_in_7_FALSE)

t.test(mean_C_in_7_TRUE$X.SNPs.BE, mean_C_in_7_FALSE$X.SNPs.BE, var.equal = TRUE)

# C in 6 bases from PAM in BE GUARDs
#stats

mean_C_in_6_TRUE <- data_clean_BE_split %>% 
filter(C_in_6 == "TRUE") %>%
as.data.frame(mean_C_in_6_TRUE)

mean_C_in_6_FALSE <- data_clean_BE_split %>%
filter(C_in_6 == "FALSE") %>%
as.data.frame(mean_C_in_6_FALSE)

t.test(mean_C_in_6_TRUE$X.SNPs.BE, mean_C_in_6_FALSE$X.SNPs.BE, var.equal = TRUE)

# C in 5 bases from PAM in BE GUARDs
#stats

mean_C_in_5_TRUE <- data_clean_BE_split %>% 
filter(C_in_5 == "TRUE") %>%
as.data.frame(mean_C_in_5_TRUE)

mean_C_in_5_FALSE <- data_clean_BE_split %>%
filter(C_in_5 == "FALSE") %>%
as.data.frame(mean_C_in_5_FALSE)

t.test(mean_C_in_5_TRUE$X.SNPs.BE, mean_C_in_5_FALSE$X.SNPs.BE, var.equal = TRUE)

# C in 4 bases from PAM in BE GUARDs
#stats

mean_C_in_4_TRUE <- data_clean_BE_split %>% 
filter(C_in_4 == "TRUE") %>%
as.data.frame(mean_C_in_4_TRUE)

mean_C_in_4_FALSE <- data_clean_BE_split %>%
filter(C_in_4 == "FALSE") %>%
as.data.frame(mean_C_in_4_FALSE)

t.test(mean_C_in_4_TRUE$X.SNPs.BE, mean_C_in_4_FALSE$X.SNPs.BE, var.equal = TRUE)

# C in 3 bases from PAM in BE GUARDs
#stats

mean_C_in_3_TRUE <- data_clean_BE_split %>% 
filter(C_in_3 == "TRUE") %>%
as.data.frame(mean_C_in_3_TRUE)

mean_C_in_3_FALSE <- data_clean_BE_split %>%
filter(C_in_3 == "FALSE") %>%
as.data.frame(mean_C_in_3_FALSE)

t.test(mean_C_in_3_TRUE$X.SNPs.BE, mean_C_in_3_FALSE$X.SNPs.BE, var.equal = TRUE)

# C in 2 bases from PAM in BE GUARDs
#stats

mean_C_in_2_TRUE <- data_clean_BE_split %>% 
filter(C_in_2 == "TRUE") %>%
as.data.frame(mean_C_in_2_TRUE)

mean_C_in_2_FALSE <- data_clean_BE_split %>%
filter(C_in_2 == "FALSE") %>%
as.data.frame(mean_C_in_2_FALSE)

t.test(mean_C_in_2_TRUE$X.SNPs.BE, mean_C_in_2_FALSE$X.SNPs.BE, var.equal = TRUE)

# C in 1 bases from PAM in BE GUARDs
#stats

mean_C_in_1_TRUE <- data_clean_BE_split %>% 
filter(C_in_1 == "TRUE") %>%
as.data.frame(mean_C_in_1_TRUE)

mean_C_in_1_FALSE <- data_clean_BE_split %>%
filter(C_in_1 == "FALSE") %>%
as.data.frame(mean_C_in_1_FALSE)

t.test(mean_C_in_1_TRUE$X.SNPs.BE, mean_C_in_1_FALSE$X.SNPs.BE, var.equal = TRUE)
```

```{r Cas9 strand stats}
# Cas9 strand stats

mean_strand_TRUE <- data_clean_Cas9_split %>% 
filter(SameStrand == "TRUE") %>%
as.data.frame(mean_strand_TRUE)

mean_strand_FALSE <- data_clean_Cas9_split %>%
filter(SameStrand == "FALSE") %>%
as.data.frame(mean_strand_FALSE)

t.test(mean_strand_TRUE$X.Indels.Cas9, mean_strand_FALSE$X.Indels.Cas9, var.equal = TRUE)

```

```{r Cas9 strand plot}
# Cas9 strand plot

plot_Cas9_Strand <- ggplot(subset(data_clean_Cas9_split, NT =="FALSE"), aes(x=SameStrand, y=X.Indels.Cas9)) +
geom_boxplot(colour = "Red") +
  theme_bw() +
geom_jitter(alpha=0.1, size = 2, width = 0.2) +
  labs(title= "Cas9: GUARD RNA strand relative to gRNA", x=NULL, y="Indels (%)") +
annotate("text", x = 1.5, y = 20, label = c("P= 0.009491")) +
annotate("segment", x = 1.1, xend = 1.9, y = 19.4, yend = 19.4, alpha = 0.7) +
coord_cartesian(ylim=c(0, 20)) +
scale_x_discrete(labels=c("FALSE" = "Opposite strand", "TRUE" = "Same strand")) + 
theme(axis.text.x = element_text(size = rel(1.2)))
plot_Cas9_Strand

```

```{r HiFiCas9 strand stats}
# HiFiCas9 strand stats

mean_strand_TRUE_HiFi <- data_clean_HiFiCas9_split %>% 
filter(SameStrand == "TRUE") %>%
as.data.frame(mean_strand_TRUE_HiFi)

mean_strand_FALSE_HiFi <- data_clean_HiFiCas9_split %>%
filter(SameStrand == "FALSE") %>%
as.data.frame(mean_strand_FALSE_HiFi)

t.test(mean_strand_TRUE_HiFi$X.Indels.HiFiCas9, mean_strand_FALSE_HiFi$X.Indels.HiFiCas9, var.equal = TRUE)

```

```{r HiFiCas9 strand plot}
# HiFiCas9 strand plot

plot_HiFiCas9_Strand <- ggplot(subset(data_clean_HiFiCas9_split, NT =="FALSE"), aes(x=SameStrand, y=X.Indels.HiFiCas9)) +
geom_boxplot(colour = "Red") +
  theme_bw() +
geom_jitter(alpha=0.1, size = 2, width = 0.2) +
  labs(title= "HiFi-Cas9: GUARD RNA strand relative to gRNA", x=NULL, y="Indels (%)") +
annotate("text", x = 1.5, y = 20, label = c("P= 0.001943")) +
annotate("segment", x = 1.1, xend = 1.9, y = 19.4, yend = 19.4, alpha = 0.7) +
coord_cartesian(ylim=c(0, 20)) +
scale_x_discrete(labels=c("FALSE" = "Opposite strand", "TRUE" = "Same strand")) + 
theme(axis.text.x = element_text(size = rel(1.2)))
plot_HiFiCas9_Strand

```

```{r BE strand stats}
# BE strand stats

mean_strand_TRUE <- data_clean_BE_split %>% 
filter(SameStrand == "TRUE") %>%
as.data.frame(mean_strand_TRUE)

mean_strand_FALSE <- data_clean_BE_split %>%
filter(SameStrand == "FALSE") %>%
as.data.frame(mean_strand_FALSE)

t.test(mean_strand_TRUE$X.SNPs.BE, mean_strand_FALSE$X.SNPs.BE, var.equal = TRUE)

```

```{r BE strand plot}
# BE strand plot

plot_BE_Strand <- ggplot(subset(data_clean_BE_split, NT =="FALSE"), aes(x=SameStrand, y=X.SNPs.BE)) +
geom_boxplot(colour = "Red") +
  theme_bw() +
geom_jitter(alpha=0.1, size = 2, width = 0.2) +
  labs(title= "BE: GUARD RNA strand relative to gRNA", x=NULL, y="BE (%)") +
annotate("text", x = 1.5, y = 98, label = c("P= 0.002458")) +
annotate("segment", x = 1.1, xend = 1.9, y = 95, yend = 95, alpha = 0.7)+
scale_x_discrete(labels=c("FALSE" = "Opposite strand", "TRUE" = "Same strand")) +
theme(axis.text.x = element_text(size = rel(1.2)))
plot_BE_Strand

```

```{r plot off-target IDs for Cas9}
# plot off-target IDs for Cas9

plot_NT_Cas9_Functional <- ggplot(data_clean_Cas9_split, aes(x=NT, y=X.Indels.Cas9, colour = Functional)) +
geom_jitter(size=1, alpha=0.8, width = 0.2, height = 0.2) + labs(title= "Cas9: Off-targets", x=NULL, y="Indels (%)") +
facet_wrap(~ID, scales ='free_y') +
scale_x_discrete(labels=c("FALSE" = "GUARD RNAs", "TRUE" = "NT controls"))+
theme(axis.text.x = element_text(angle = 90))
plot_NT_Cas9_Functional

```

```{r plot subset of off-target IDs for Cas9}
# plot subset of off-target IDs for Cas9

data_select_IDs_Cas9 <- data_clean_Cas9_split %>% 
filter(ID =="OT0006-HCN1_EMX1 \n Mismatches =  4" | ID =="OT0005-RPTOR_FANCF \n Mismatches =  4" | ID == "OT0002-LINC00608_EMX1 \n Mismatches =  4" | ID == "OT0001_HEKsite1 \n Mismatches =  3" | ID == "OT0009_HBB \n Mismatches =  3")

plot_NT_Cas9_Functional_subset <- ggplot(data_select_IDs_Cas9, aes(x=NT, y=X.Indels.Cas9, colour = Functional)) +
geom_jitter(size=1.5, alpha=0.8, width = 0.15, height = 0.1) + labs(title= "Cas9: Off-targets", x=NULL, y="Indels (%)") +
facet_wrap(~ID, scales ='free', nrow=1) +
scale_x_discrete(labels=c("FALSE" = "GUARD RNAs", "TRUE" = "NT controls"))+
theme(axis.text.x = element_text(angle = 90))
plot_NT_Cas9_Functional_subset

```

```{r plot off-target IDs for HiFi Cas9}
# plot off-target IDs for HiFiCas9

plot_NT_HiFiCas9_Functional <- ggplot(data_clean_HiFiCas9_split, aes(x=NT, y=X.Indels.HiFiCas9, colour = Perfect)) +
geom_jitter(size=1, alpha=0.8, width = 0.2, height = 0.2) + labs(title= "HiFi-Cas9: Off-targets", x=NULL, y="Indels (%)") +
facet_wrap(~ID, scales ='free_y') +
scale_x_discrete(labels=c("FALSE" = "GUARD RNAs", "TRUE" = "NT controls"))+
theme(axis.text.x = element_text(angle = 90))
plot_NT_HiFiCas9_Functional

```

```{r plot subset of off-target IDs for HiFiCas9}
# plot subset of off-target IDs for HiFiCas9

data_select_IDs_HiFiCas9 <- data_clean_HiFiCas9_split %>% 
filter(ID =="OT0006-HCN1_EMX1 \n Mismatches =  4" | ID =="OT0005-RPTOR_FANCF \n Mismatches =  4" | ID == "OT0002-LINC00608_EMX1 \n Mismatches =  4" | ID == "OT0001_HEKsite1 \n Mismatches =  3" | ID == "OT0009_HBB \n Mismatches =  3")

plot_NT_HiFiCas9_Functional_subset <- ggplot(data_select_IDs_HiFiCas9, aes(x=NT, y=X.Indels.HiFiCas9, colour = Perfect)) +
geom_jitter(size=1.5, alpha=0.8, width = 0.15, height = 0.1) + labs(title= "HiFi-Cas9: Off-targets", x=NULL, y="Indels (%)") +
facet_wrap(~ID, scales ='free', nrow=1) +
scale_x_discrete(labels=c("FALSE" = "GUARD RNAs", "TRUE" = "NT controls"))+
theme(axis.text.x = element_text(angle = 90))
plot_NT_HiFiCas9_Functional_subset

```

```{r plot off-target IDs for BE}
# plot off-target IDs for BE

plot_NT_BE_Functional <- ggplot(data_clean_BE_split, aes(x=NT, y=X.SNPs.BE, colour = Functional)) +
geom_jitter(size=1, alpha=0.8, width = 0.2, height = 0.2) + labs(title= "BE: Off-targets", x=NULL, y="BE (%)") +
facet_wrap(~ID, scales ='free_y')+
scale_x_discrete(labels=c("FALSE" = "GUARD RNAs", "TRUE" = "NT controls")) +
theme(axis.text.x = element_text(angle = 90))
plot_NT_BE_Functional

```

```{r plot subset of off-target IDs for BE}
# plot subset of off-target IDs for BE

data_select_IDs_BE <- data_clean_BE_split %>% 
filter(ID =="OT0006-HCN1_EMX1 \n Mismatches =  4" | ID =="OT0005-RPTOR_FANCF \n Mismatches =  4" | ID == "OT0002-LINC00608_EMX1 \n Mismatches =  4" | ID == "OT0001_HEKsite1 \n Mismatches =  3" | ID == "OT0009_HBB \n Mismatches =  3")

plot_NT_BE_Functional_subset <- ggplot(data_select_IDs_BE, aes(x=NT, y=X.SNPs.BE, colour = Functional)) +
geom_jitter(size=1.5, alpha=0.8, width = 0.15, height = 0.1) + labs(title= "BE: Off-targets", x=NULL, y="BE (%)") +
facet_wrap(~ID, scales ='free', nrow=1) +
scale_x_discrete(labels=c("FALSE" = "GUARD RNAs", "TRUE" = "NT controls"))+
theme(axis.text.x = element_text(angle = 90))
plot_NT_BE_Functional_subset

```

```{r NGG Cas9 stats}
# NGG Cas9 stats

x <- data_clean_Cas9_split %>% group_by(Functional, NGG) %>% select(NGG, Functional)
fisher.test(table(x))

```

```{r NGG Cas9 plot}
# NGG Cas9 plot

plot_NGG_Cas9 <- ggplot(data_clean_Cas9_split, aes(x=NGG, fill = Functional)) +
geom_bar(stat="count", position = "fill") + labs(title= "Cas9: PAM", x=NULL, y="Proportion of GUARDs") +
annotate("text", x = 1.5, y = 0.95, label = c("P = 2.2 x 10-16")) +
annotate("segment", x = 1.1, xend = 1.9, y = 0.92, yend = 0.92, alpha = 0.7) +
scale_x_discrete(labels=c("FALSE" = "NAG PAM", "TRUE" = "NGG PAM")) +
theme(axis.text.x = element_text(size = rel(1.2)))
plot_NGG_Cas9

```

```{r NGG HiFiCas9 stats}
# NGG HiFiCas9 stats

x <- data_clean_HiFiCas9_split %>% group_by(Perfect, NGG) %>% select(NGG, Perfect)
fisher.test(table(x))

```

```{r NGG HiFiCas9 plot}
# NGG HiFiCas9 plot

plot_NGG_HiFiCas9 <- ggplot(data_clean_HiFiCas9_split, aes(x=NGG, fill = Perfect)) +
geom_bar(stat="count", position = "fill") + labs(title= "HiFi-Cas9: PAM", x=NULL, y="Proportion of GUARDs") +
annotate("text", x = 1.5, y = 0.95, label = c("P = 5.6 x 10-5")) +
annotate("segment", x = 1.1, xend = 1.9, y = 0.92, yend = 0.92, alpha = 0.7) +
scale_x_discrete(labels=c("FALSE" = "NAG PAM", "TRUE" = "NGG PAM")) +
theme(axis.text.x = element_text(size = rel(1.2)))
plot_NGG_HiFiCas9

```

```{r}
# OffGuideOverlap functional vs non-functional Cas9
#stats

mean_nonfunctional <- data_clean_Cas9_split %>% group_by(Functional) %>% filter(Functional == "FALSE")
mean_functional <- data_clean_Cas9_split %>% group_by(Functional) %>% filter(Functional == "TRUE")

t.test(mean_nonfunctional$OffGuidePAMOverlap, mean_functional$OffGuidePAMOverlap, var.equal = TRUE)

```

```{r}
#plot Cas9 OffGuideOverlap

plot_PAMOverlap_Cas9 <- ggplot(subset(data_clean_Cas9_split, NT == "FALSE"), aes(x= Functional, y=OffGuidePAMOverlap)) +
geom_boxplot(colour = "Red") + 
theme_bw() +
geom_jitter(alpha=0.1, size = 2, width = 0.2) +
labs(title= "Cas9: PAM overlap", x=NULL, y="Overlap (bp)") +
annotate("text", x = 1.5, y = 20, label = c("P = 1.282 x 10-5")) +
annotate("segment", x = 1.1, xend = 1.9, y = 19.5, yend = 19.5, alpha = 0.7) +
scale_x_discrete(labels=c("FALSE" = "Non-functional \n GUARD RNAs", "TRUE" = "Functional \n GUARD RNAs")) +
theme(axis.text.x = element_text(size = rel(1.2)))+
ylim(0,20)
plot_PAMOverlap_Cas9

```

```{r}
# OffGuideOverlap functioal vs non-functional Cas9
#stats

mean_nonfunctional <- data_clean_Cas9_split %>% group_by(Functional) %>% filter(Functional == "FALSE")
mean_functional <- data_clean_Cas9_split %>% group_by(Functional) %>% filter(Functional == "TRUE")

t.test(mean_nonfunctional$OffGuideSeedOverlap, mean_functional$OffGuideSeedOverlap, var.equal = TRUE)

```

```{r}
#plot Cas9 OffGuideOverlap

plot_OffGuideSeedOverlap_Cas9 <- ggplot(subset(data_clean_Cas9_split, NT == "FALSE"), aes(x= Functional, y=OffGuideSeedOverlap)) +
geom_boxplot(colour = "Red") + 
theme_bw() +
geom_jitter(alpha=0.1, size = 2, width = 0.2) +
labs(title= "Cas9: Seed region overlap", x=NULL, y="Overlap (bp)") +
annotate("text", x = 1.5, y = 20, label = c("P = 0.009682")) +
annotate("segment", x = 1.1, xend = 1.9, y = 19.5, yend = 19.5, alpha = 0.7)+
scale_x_discrete(labels=c("FALSE" = "Non-functional \n GUARD RNAs", "TRUE" = "Functional \n GUARD RNAs")) +
theme(axis.text.x = element_text(size = rel(1.2)))+
ylim(0, 20)
plot_OffGuideSeedOverlap_Cas9

```

```{r}
# OffGuideOverlap functioal vs non-functional Cas9
#stats

mean_nonfunctional <- data_clean_Cas9_split %>% group_by(Functional) %>% filter(Functional == "FALSE")
mean_functional <- data_clean_Cas9_split %>% group_by(Functional) %>% filter(Functional == "TRUE")

t.test(mean_nonfunctional$OffGuideOverlap, mean_functional$OffGuideOverlap, var.equal = TRUE)

```

```{r}
#plot Cas9 OffGuideOverlap

plot_OffGuideOverlap_Cas9 <- ggplot(subset(data_clean_Cas9_split, NT == "FALSE"), aes(x= Functional, y=OffGuideOverlap)) +
geom_boxplot(colour = "Red") + 
theme_bw() +
geom_jitter(alpha=0.1, size =2, width = 0.2) +
labs(title= "Cas9: gRNA overlap", x=NULL, y="Overlap (bp)") +
annotate("text", x = 1.5, y = 20, label = c("P = 0.04083")) +
annotate("segment", x = 1.1, xend = 1.9, y = 19.5, yend = 19.5, alpha = 0.7) +
scale_x_discrete(labels=c("FALSE" = "Non-functional \n GUARD RNAs", "TRUE" = "Functional \n GUARD RNAs")) +
theme(axis.text.x = element_text(size = rel(1.2))) +
ylim(0, 20)
plot_OffGuideOverlap_Cas9

```

```{r}
# OffGuideOverlap functional vs non-functional HiFiCas9
#stats

mean_nonfunctionalHiFi <- data_clean_HiFiCas9_split %>% group_by(Perfect) %>% filter(Perfect == "FALSE")
mean_functionalHiFi <- data_clean_HiFiCas9_split %>% group_by(Perfect) %>% filter(Perfect == "TRUE")

t.test(mean_nonfunctionalHiFi$OffGuidePAMOverlap, mean_functionalHiFi$OffGuidePAMOverlap, var.equal = TRUE)

```

```{r}
#plot HiFiCas9 OffGuideOverlap

plot_PAMOverlap_HiFiCas9 <- ggplot(subset(data_clean_HiFiCas9_split, NT == "FALSE"), aes(x= Perfect, y=OffGuidePAMOverlap)) +
geom_boxplot(colour = "Red") + 
theme_bw() +
geom_jitter(alpha=0.1, size = 2, width = 0.2) +
labs(title= "Cas9: PAM overlap", x=NULL, y="Overlap (bp)") +
annotate("text", x = 1.5, y = 20, label = c("P = N.S")) +
annotate("segment", x = 1.1, xend = 1.9, y = 19.5, yend = 19.5, alpha = 0.7) +
scale_x_discrete(labels=c("FALSE" = "> 0.5 % Indels \n GUARD RNAs", "TRUE" = "< 0.5 % Indels \n GUARD RNAs")) +
theme(axis.text.x = element_text(size = rel(1.2)))+
ylim(0,20)
plot_PAMOverlap_HiFiCas9

```

```{r}

# OffGuideOverlap functioal vs non-functional HiFiCas9
#stats

mean_nonfunctionalHiFi <- data_clean_HiFiCas9_split %>% group_by(Perfect) %>% filter(Perfect == "FALSE")
mean_functionalHiFi <- data_clean_HiFiCas9_split %>% group_by(Perfect) %>% filter(Perfect == "TRUE")

t.test(mean_nonfunctionalHiFi$OffGuideSeedOverlap, mean_functionalHiFi$OffGuideSeedOverlap, var.equal = TRUE)

```

```{r}
#plot HiFiCas9 OffGuideOverlap

plot_OffGuideSeedOverlap_HiFiCas9 <- ggplot(subset(data_clean_HiFiCas9_split, NT == "FALSE"), aes(x= Perfect, y=OffGuideSeedOverlap)) +
geom_boxplot(colour = "Red") + 
theme_bw() +
geom_jitter(alpha=0.1, size = 2, width = 0.2) +
labs(title= "HiFi-Cas9: Seed region overlap", x=NULL, y="Overlap (bp)") +
annotate("text", x = 1.5, y = 20, label = c("P = 0.0153")) +
annotate("segment", x = 1.1, xend = 1.9, y = 19.5, yend = 19.5, alpha = 0.7)+
scale_x_discrete(labels=c("FALSE" = ">0.5 % Indels \n GUARD RNAs", "TRUE" = "<0.5 % Indels \n GUARD RNAs")) +
theme(axis.text.x = element_text(size = rel(1.2)))+
ylim(0, 20)
plot_OffGuideSeedOverlap_HiFiCas9

```

```{r}
# OffGuideOverlap functioal vs non-functional HiFiCas9
#stats

mean_nonfunctionalHiFi <- data_clean_HiFiCas9_split %>% group_by(Perfect) %>% filter(Perfect == "FALSE")
mean_functionalHiFi <- data_clean_HiFiCas9_split %>% group_by(Perfect) %>% filter(Perfect == "TRUE")

t.test(mean_nonfunctionalHiFi$OffGuideOverlap, mean_functionalHiFi$OffGuideOverlap, var.equal = TRUE)

```

```{r}
#plot HiFiCas9 OffGuideOverlap

plot_OffGuideOverlap_HiFiCas9 <- ggplot(subset(data_clean_HiFiCas9_split, NT == "FALSE"), aes(x= Perfect, y=OffGuideOverlap)) +
geom_boxplot(colour = "Red") + 
theme_bw() +
geom_jitter(alpha=0.1, size =2, width = 0.2) +
labs(title= "HiFi-Cas9: gRNA overlap", x=NULL, y="Overlap (bp)") +
annotate("text", x = 1.5, y = 20, label = c("P = 0.01983")) +
annotate("segment", x = 1.1, xend = 1.9, y = 19.5, yend = 19.5, alpha = 0.7) +
scale_x_discrete(labels=c("FALSE" = ">0.5 % Indels \n GUARD RNAs", "TRUE" = "<0.5 % Indels \n GUARD RNAs")) +
theme(axis.text.x = element_text(size = rel(1.2))) +
ylim(0, 20)
plot_OffGuideOverlap_HiFiCas9

```

```{r stats on functional GUARD RNA numbers}
# stats on functional GUARD RNA numbers

# Check the number of GUARDs that are functional and the number of off-targets that have protection by functional GUARDs

# Cas9 Functional
data_clean_Cas9_split %>%
group_by(Number, ID, Condition, Functional) %>%
filter(Condition == "Dox.1" & NT == "FALSE") %>%
tally() %>%
count(n)

data_clean_Cas9_split %>%
group_by(Number, ID, Condition, Guide, Functional, NT, mean_NT, X.Indels.Cas9) %>%
filter(Functional == "TRUE") %>%
filter(Condition == "Dox.1"& NT == "FALSE") %>%
tally() %>%
count(n)

# Check the number of GUARDs that are functional and block editing to less than < 0.5 %

data_clean_Cas9_split %>%
group_by(Number, ID, Condition, Guide, Functional, NT, mean_NT, X.Indels.Cas9) %>%
filter(Functional == "TRUE") %>%
filter(Condition == "Dox.1" & NT == "FALSE") %>%
filter(X.Indels.Cas9 < 0.5) %>% 
tally() %>%
count(n)

# Cas9
data_clean_Cas9_split %>%
group_by(Number, ID, Condition, Guide, Functional, NT, mean_NT, X.Indels.Cas9) %>%
filter(Condition == "Dox.2" & NT == "FALSE") %>%
tally() %>%
count(n)

data_clean_Cas9_split %>%
group_by(Number, ID, Condition, Guide, Functional, NT, mean_NT, X.Indels.Cas9) %>%
filter(Functional == "TRUE") %>%
filter(Condition == "Dox.2" & NT == "FALSE") %>%
tally() %>%
count(n)

# Check the number of GUARDs that are functional and block editing to less than < 0.5 %

data_clean_Cas9_split %>%
group_by(Number, ID, Condition, Guide, Functional, NT, mean_NT, X.Indels.Cas9) %>%
filter(Functional == "TRUE") %>%
filter(Condition == "Dox.2" & NT == "FALSE") %>%
filter(X.Indels.Cas9 < 0.5) %>% 
tally() %>%
count(n)


# HiFiCas9

# Cas9 Functional
data_clean_HiFiCas9_split %>%
group_by(Number, ID, Condition, Functional) %>%
filter(Condition == "Dox.1" & NT == "FALSE") %>%
tally() %>%
count(n)

data_clean_HiFiCas9_split %>%
group_by(Number, ID, Condition, Guide, Functional, NT, mean_NT, X.Indels.Cas9) %>%
filter(Functional == "TRUE") %>%
filter(Condition == "Dox.1"& NT == "FALSE") %>%
tally() %>%
count(n)

# Check the number of GUARDs that block editing to less than < 0.5 %

data_clean_HiFiCas9_split %>%
group_by(Number, ID, Condition, Guide, Functional, NT, mean_NT, X.Indels.HiFiCas9) %>%
#filter(Functional == "TRUE") %>%
filter(Condition == "Dox.1" & NT == "FALSE") %>%
filter(X.Indels.HiFiCas9 < 0.5) %>% 
tally() %>%
count(n)

# HiFiCas9
data_clean_HiFiCas9_split %>%
group_by(Number, ID, Condition, Guide, Functional, NT, mean_NT, X.Indels.HiFiCas9) %>%
filter(Condition == "Dox.2" & NT == "FALSE") %>%
tally() %>%
count(n)

data_clean_HiFiCas9_split %>%
group_by(Number, ID, Condition, Guide, Functional, NT, mean_NT, X.Indels.HiFiCas9) %>%
filter(Functional == "TRUE") %>%
filter(Condition == "Dox.2" & NT == "FALSE") %>%
tally() %>%
count(n)

# Check the number of GUARDs that block editing to less than < 0.5 %

data_clean_HiFiCas9_split %>%
group_by(Number, ID, Condition, Guide, Functional, NT, mean_NT, X.Indels.HiFiCas9) %>%
#filter(Functional == "TRUE") %>%
filter(Condition == "Dox.2" & NT == "FALSE") %>%
filter(X.Indels.HiFiCas9 < 0.5) %>% 
tally() %>%
count(n)



# BE
data_clean_BE_split %>%
group_by(Number, ID, Condition, Guide, Functional, NT, mean_NT, X.SNPs.BE) %>%
filter(Condition == "Dox.1" & NT == "FALSE") %>%
tally() %>%
count(n)

data_clean_BE_split %>%
group_by(Number, ID, Condition, Guide, Functional, NT, mean_NT, X.SNPs.BE) %>%
filter(Functional == "TRUE") %>%
filter(Condition == "Dox.1" & NT == "FALSE") %>%
tally() %>%
count(n)

# Check the number of GUARDs that are functional and block editing to less than < 0.5 %

data_clean_BE_split %>%
group_by(Number, ID, Condition, Guide, Functional, NT, mean_NT, X.SNPs.BE) %>%
filter(Functional == "TRUE") %>%
filter(Condition == "Dox.1" & NT == "FALSE") %>%
filter(X.SNPs.BE < 0.5) %>% 
tally() %>%
count(n)

# BE
data_clean_BE_split %>%
group_by(Number, ID, Condition, Guide, Functional, NT, mean_NT, X.SNPs.BE) %>%
filter(Condition == "Dox.2" & NT == "FALSE") %>%
tally() %>%
count(n)

data_clean_BE_split %>%
group_by(Number, ID, Condition, Guide, Functional, NT, mean_NT, X.SNPs.BE) %>%
filter(Functional == "TRUE") %>%
filter(Condition == "Dox.2" & NT == "FALSE") %>%
tally() %>%
count(n)

# Check the number of GUARDs that are functional and block editing to less than < 0.5 %

data_clean_BE_split %>%
group_by(Number, ID, Condition, Guide, Functional, NT, mean_NT, X.SNPs.BE) %>%
filter(Functional == "TRUE") %>%
filter(Condition == "Dox.2" & NT == "FALSE") %>%
filter(X.SNPs.BE < 0.5) %>% 
tally() %>%
count(n)

```

```{r count the number of functional GUARDs with GC-content, Cas9}
# count the number of functional GUARDs with GC-content, Cas9

plot_OffGuideGC_Cas9 <- ggplot(subset(data_clean_Cas9_split, NT == "FALSE"), aes(x = GuardGC, fill=Functional)) +
geom_bar(stat="count", position = "fill") + labs(title= "Cas9: GUARD RNA GC content", x="GUARD RNA GC content", y="Proportion of GUARDs")
plot_OffGuideGC_Cas9

```

```{r count the number of functional GUARDs with GC-content, HiFiCas9}
# count the number of functional GUARDs with GC-content, Cas9

plot_OffGuideGC_HiFiCas9 <- ggplot(subset(data_clean_HiFiCas9_split, NT == "FALSE"), aes(x = GuardGC, fill=Perfect)) +
geom_bar(stat="count", position = "fill") + labs(title= "HiFi-Cas9: GUARD RNA GC content", x="GUARD RNA GC content", y="Proportion of GUARDs")
plot_OffGuideGC_HiFiCas9

```

```{r proportion of SNPs that are C-T and G-A mutations}
# proportion of SNPs that are C-T and G-A mutations

data_CTGA <- data_clean_BE_split %>% group_by(Number) %>% mutate(CT_GA = sum(X.CT.BE, X.GA.BE))

plot_CT_BE <- ggplot(data_CTGA, aes(x = CT_GA, y= X.SNPs.BE)) +
geom_jitter() + labs(title= "Base Edits: C to T / G to A (%)", x="CT + GA mutations (%)", y="Total SNPs (%)") +
scale_colour_continuous(low = "Blue", high="Red")
plot_CT_BE

```

```{r plot MFAP1 validation set for Cas9}
# plot MFAP1 validation set for Cas9
#install.packages("ggrepel")
library(ggrepel)
data_MFAP1_Cas9 <- data_clean_Cas9_split %>% 
filter(ID =="OT0011-MFAP1_EMX1 \n Mismatches =  4")

plot_data_MFAP1_Cas9 <- ggplot(data_MFAP1_Cas9, aes(x=NT, y=X.Indels.Cas9, colour = Functional)) +
geom_jitter(size=1.5, alpha=0.8, width = 0.1, height = 0.1) + labs(title= "Cas9: Off-targets", x=NULL, y="Indels (%)") +
scale_x_discrete(labels=c("FALSE" = "GUARD RNAs", "TRUE" = "NT controls"))+
facet_wrap(~ID, scales ='free_y')+
theme(axis.text.x = element_text(angle = 90)) +
geom_text_repel(aes(label=ifelse(GUARDWithG=="GTAGATGGTTGGGTAG", "GUARD v2", "")), nudge_y=0.2, nudge_x =0.2) +
geom_text_repel(aes(label=ifelse(GUARDWithG=="GTAAGCAGAAGAAGAA", "GUARD v1", "")), nudge_y=0.2, nudge_x =0.2)
plot_data_MFAP1_Cas9

```

##################################################################################################################################################################################
